### PR TITLE
Feature/server 2531 delete push notification keys

### PR DESCRIPTION
**Pull request by LumbardhAgaj**
User explicitly deletes a push notification key. Upon doing so, he must confirm his action using the dialog.

fix:
 - updating rate overwrite error in the server-side

### CODE REVIEW

File: plugins/push/api/api-push.js [hunk 1]
```diff
@@ -181,11 +181,11 @@ module.exports.onAppPluginsUpdate = async({params, app, config}) => {
         }
 
         if (config.rate.period) {
-            update.$set = {'plugins.push.rate.period': config.rate.period};
+            update.$set = Object.assign(update.$set, {'plugins.push.rate.period': config.rate.period});
```
Comment (iartem) [inline plugins/push/api/api-push.js:184]: good catch

File: plugins/push/api/api-push.js [hunk 2]
```diff
@@ -181,11 +181,11 @@ module.exports.onAppPluginsUpdate = async({params, app, config}) => {
...
             pushcfg.rate.period = config.rate.period;
         }
         else {
-            update.$unset = {'plugins.push.rate.period': 1};
+            update.$unset = Object.assign(update.$unset, {'plugins.push.rate.period': 1});
```
Comment (iartem) [inline plugins/push/api/api-push.js:188]: bad catch XD 

`$unset` might not exist here

File: plugins/push/api/api-push.js (hunk 1 above)
Comment (iartem) [reply plugins/push/api/api-push.js:184]: ok, it's a bad catch as well lol as `$set` might not be there either

File: plugins/push/frontend/public/templates/push-notification-app-level-config.html [hunk 3]
```diff
@@ -59,4 +70,14 @@
...
+    <cly-confirm-dialog 
+        title="Delete key"
+        dialogType="danger"
+        saveButtonLabel="I understand, delete this key" 
+        cancelButtonLabel="Cancel" 
+        :visible.sync="isDeleteKeyDialogVisible"
+        @cancel="onDeleteKeyCancel" 
+        @confirm="onDeleteKeyConfirm">
+    </cly-confirm-dialog>
+</cly-confirm-dialog >
```
Comment (can-keklik) [inline plugins/push/frontend/public/templates/push-notification-app-level-config.html:82]: You can use CountlyHelpers.confirm unless you have a specific reason not to do so :slightly_smiling_face:

**Review by can-keklik — APPROVED** [review]
I left just one comment. Other than that, it is up to @iartem :slightly_smiling_face:

File: plugins/push/frontend/public/templates/push-notification-app-level-config.html (hunk 3 above)
Comment (LumbardhAgaj) [reply plugins/push/frontend/public/templates/push-notification-app-level-config.html:82]: yes I will. It slipped my mind.

**Pull request merged**